### DISCUSSION TITLE
How does the mass fraction in the Fluid Mass Governing equation get updated in PorousFlow?

### DISCUSSION THREAD

**Discussion by steam-simulator** [idaholab/moose#26280] [Q&A Modules: Porous Flow] [answered]
Hello,
I wonder how the mass fraction gets updated in the `PorousFlowMassFraction` material after its initial values get called into this material. There is no equation in the `PorousFlowMassFraction` documentation in this regard. I’d be grateful if someone could help out. 
Thanks.

**Comment by cpgr** [comment, marked as answer]
That material doesn't really do anything except put all of the mass fractions in a multi-dimensional array in the correct positions so every other object (like the kernels) can access mass fractions of each fluid component in each fluid phase. So the mass fraction won't change unless it is a nonlinear variable, or if it is computed in some other object (for example, the brine-CO2 stuff computes the mass fractions of water and CO2 in both phases depending on the conditions).

**Reply by steam-simulator** [reply]
Hello,

The answer to my previous post #26168 suggests that the mass fraction of the primary chemical species in PorousFlow chemistry also represents the primary species concentration. It seems PorousFlow requires primary species concentration to be very small (preferably a minimum of 1 [units]). Correct? 
I expected my primary concentrations to be about 6 [units], which my dissolution rate correctly gave me. As a result, the water mass fraction becomes exceedingly negative (since `\chi^{water}=1-\chi^{all_components}`. Is there a way I can normalize my primary concentration, hence the mass fraction, so that I obtain a positive water mass fraction?
Also, if I’m using mass fraction as a non-linear variable for PorousFlow chemistry, how does Porousflow compute the mass fraction’s evolution? The answer to this question will help me know how Porousflow is computing my primary concentrations. Does it depend on the reaction rate? Thanks.
